### PR TITLE
Identicons

### PR DESCRIPTION
**Pull request by Agupane**
## Description
- Adds identicon component

### CODE REVIEW

File: src/dataDisplay/Identicon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+import makeBlockie from 'ethereum-blockies-base64';
+import { RefObject, useEffect, useState } from 'react';
+
+type Props = {
+  address: string;
+  diameter: number;
```
Comment (nicosampler) [inline src/dataDisplay/Identicon/index.tsx:8]: Diameter values should be handled through the theme, as we do for Text size for example.

File: src/dataDisplay/Identicon/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  title: 'Data Display/Identicon',
+  component: Identicon,
+  parameters: {
+    componentSubtitle: 'Identicon Component.'
+  }
+};
+
+export const sizes = () => {
+  return (
+    <>
```
Comment (nicosampler) [inline src/dataDisplay/Identicon/index.stories.tsx:15]: I think we can export `<Identicon />` directly, without a need of using `<></>`

File: src/dataDisplay/Identicon/index.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  image.height = diameter;
+  image.width = diameter;
+  image.style.borderRadius = `${diameter / 2}px`;
+
+  return image;
+};
+
+const Identicon = (props: Props) => {
+  const { address, diameter } = props;
+  const style: StyleProps = getStyleFrom(diameter);
```
Comment (nicosampler) [inline src/dataDisplay/Identicon/index.tsx:36]: No need to define `style: StyledProps`, the type it's inferred.

File: src/dataDisplay/Identicon/index.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  const [identicon, setIdenticon] = useState<RefObject<HTMLImageElement> | null>(null);
+
+  useEffect(() => {
+    setIdenticon(React.createRef<HTMLImageElement>());
+  }, []);
+
+  useEffect(() => {
+    const image = generateBlockieIdenticon(address, diameter);
+
+    if (!identicon || !identicon.current) {
```
Comment (nicosampler) [inline src/dataDisplay/Identicon/index.tsx:47]: you can simplify the condition in this way: `if(!identicon?.current)`

Also, you could move this block to the first sentence, so if it is not identicon attached, the image shouldn't be created.

**Comment by Agupane** [comment]
Hey @nicosampler I simplified the component, should be better now , let me know what do you think :)

**Comment by nicosampler** [comment]
great, I think we should change this: https://github.com/gnosis/safe-react-components/pull/17/files/ca3b021928af9b8d8028f2f8153dcb38ffddf189..d9afd368f19d93d35dbb20a74c06ea6324ffd189#r438738235. We should try to avoid using numbers for sizes as much as possible.

**Comment by Agupane** [comment]
> great, I think we should change this: https://github.com/gnosis/safe-react-components/pull/17/files/ca3b021928af9b8d8028f2f8153dcb38ffddf189..d9afd368f19d93d35dbb20a74c06ea6324ffd189#r438738235. We should try to avoid using numbers for sizes as much as possible.

It's better now

File: src/theme.ts [hunk 5]
```diff
@@ -109,6 +109,13 @@ const theme = {
       sm: '16',
       md: '24'
     }
+  },
+  identicon: {
+    diameter: {
+      sm: '16',
+      md: '32',
+      lg: '60px'
```
Comment (nicosampler) [inline src/theme.ts:117]: why this one with px and the others without it? also, could we name it `size` instead, to be consistent with the other components?

**Comment by Agupane** [comment]
Should be ok now, please check this out @nicosampler

**Pull request merged**